db/hybrid_logical_vector.go

```diff
@@ -111,22 +198,88 @@ func (hlv *HybridLogicalVector) ExtractCurrentVersionFromHLV() *Version {
+	var hlvJSON HybridLogicalVectorJSON
+	var alias HLVAlias
+	alias = (HLVAlias)(hlv)
+	hlvJSON.HLVAlias = &alias
+	return base.JSONMarshal(hlvJSON)
+}
+
+func (ver HLVVersions) MarshalJSON() ([]byte, error) {
+	var verList []string
+	if len(ver) > 0 {
+		verList = VersionsToDeltas(ver)
+		return base.JSONMarshal(verList)
+	}
+	return nil, nil
+}
+
+func (ver *HLVVersions) UnmarshalJSON(inputjson []byte) error {
+
+	var verList []string
+	err := base.JSONUnmarshal(inputjson, &verList)
+	if err != nil {
+		return err
+	}
+	verMap, deltaErr := PersistedDeltasToMap(verList)
+	if deltaErr != nil {
+		return deltaErr // need typed erro to assert on
+	}
+	*ver = verMap
+	return nil
+}
+
+func (hlv *HybridLogicalVector) UnmarshalJSON(data []byte) error {
+
+	var hlvJSON *HybridLogicalVectorJSON
+	err := base.JSONUnmarshal(data, &hlvJSON)
+	if err != nil {
+		return err
+	}
+	if hlvJSON.HLVAlias != nil {
+		*hlv = HybridLogicalVector(*hlvJSON.HLVAlias)
+	}
+	return nil
+}
+
+// HybridLogicalVector is the in memory format for the hLv.
 type HybridLogicalVector struct {
-	CurrentVersionCAS string            `json:"cvCas,omitempty"`     // current version cas (or cvCAS) stores the current CAS in little endian hex format at the time of replication
-	ImportCAS         string            `json:"importCAS,omitempty"` // Set when an import modifies the document CAS but preserves the HLV (import of a version replicated by XDCR)
-	SourceID          string            `json:"src"`                 // source bucket uuid in (base64 encoded format) of where this entry originated from
-	Version           string            `json:"ver"`                 // current cas in little endian hex format of the current version on the version vector
-	MergeVersions     map[string]string `json:"mv,omitempty"`        // map of merge versions for fast efficient lookup
-	PreviousVersions  map[string]string `json:"pv,omitempty"`        // map of previous versions for fast efficient lookup
+	CurrentVersionCAS string `json:"cvCas,omitempty"`     // current version cas (or cvCAS) stores the current CAS in little endian hex format at the time of replication
+	ImportCAS         string `json:"importCAS,omitempty"` // Set when an import modifies the document CAS but preserves the HLV (import of a version replicated by XDCR)
+	SourceID          string `json:"src"`                 // source bucket uuid in (base64 encoded format) of where this entry originated from
+	Version           string `json:"ver"`                 // current cas in little endian hex format of the current version on the version vector
+	PrevVersions             // map of merge versions for fast efficient lookup
+	MVersions                // map of previous versions for fast efficient lookup
+}
+
+// PersistedHLV is the version of the version vector that is persisted to the bucket, pv and mv will be lists of source version pairs in the bucket
+type PersistedHLV struct {
```

Comment:
With the updated JSONMarshal approach, I was anticipating we'd be able to remove PersistedHLV and just use HybridLogicalVector.  I also don't think the aliasing (HLVAlias) is needed with this approach.
Have a look at https://go.dev/play/p/aYbiN9q-aXQ for the basic approach I was imagining, and let me know if there's something you don't think will work with that approach.